packages/edit-post/src/store/actions.js

```diff
@@ -436,3 +440,47 @@ export function setIsEditingTemplate( value ) {
 		value,
 	};
 }
+
+export function* __unstableSwitchToEditingMode( shouldCreateTemplate = false ) {
+	if ( shouldCreateTemplate ) {
+		const templateContent = [
+			createBlock( 'core/post-title' ),
+			createBlock( 'core/post-content' ),
+		];
+		const template = {
+			slug: 'custom-template-' + uuid(),
+			content: serialize( templateContent ),
+			title: 'Custom Template',
```

Comment:
To avoid having all custom templates have the same name by default, I think we should do something similar to template parts and reusable block: show a modal when clicking the "Create custom template" link and ask the user to fill a title.

Reply:
A title would be useful in identifying the templates 👍